wpiutil/src/main/native/cpp/SafeThread.cpp

```diff
@@ -43,7 +63,11 @@ void detail::SafeThreadOwnerBase::Start(std::shared_ptr<SafeThreadBase> thr) {
   if (auto thr = m_thread.lock()) {
     return;
   }
-  m_stdThread = std::thread([=] { thr->Main(); });
+  m_stdThread = std::thread([=] {
+    void* opaque = (gOnSafeThreadStart.load())();
+    thr->Main();
+    (gOnSafeThreadEnd.load())(opaque);
```

Comment:
Should the end callback be loaded ahead of time? Otherwise they could mismatch.

If so, they should also be behind a single synchronisation primitive.

Reply:
The intent is that you call it once at the beginning of the program and you never touch it again. If you call it after the first SafeThread has started, game's over. Probably should add a check to that effect.